docker.gradle

```diff
@@ -22,14 +23,21 @@ ext {
     dockerTimeout = project.hasProperty('dockerTimeout') ? dockerTimeout.toInteger() : 840
     dockerRetries = project.hasProperty('dockerRetries') ? dockerRetries.toInteger() : 3
     dockerBinary = project.hasProperty('dockerBinary') ? [dockerBinary] : ['docker']
+    dockerBuildArgs = project.hasProperty('dockerBuildArgs') ? dockerBuildArgs : ''
 }
 
 if(project.hasProperty('dockerHost')) {
     dockerBinary += ['--host', project.dockerHost]
 }
 
 task distDocker << {
-    def cmd = dockerBinary + ['build', '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    def cmd
+    if (dockerBuildArgs != '') {
+        cmd = dockerBinary + ['build', '--build-arg', dockerBuildArgs, '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    }
+    else {
+        cmd = dockerBinary + ['build', '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    }
```

Comment:
Currently, the CLI build version is baked into the Go CLI at build time.  No need to read an external props file for that value.  If I understand correctly, the whisk.properties is the basis for the default.props used by the python CLI.  If so, perhaps the Go CLI can leverage whisk.properties as well - just during build time